src/common/contexts/provider.tsx

```diff
@@ -49,44 +74,118 @@ declare global {
 
 interface ProviderContextProviderProps {
   children: React.ReactNode;
+  networks: ChainInfoObject[];
+  defaultChainId: ChainId;
 }
 
-export const ProviderContextProvider: FunctionComponent<ProviderContextProviderProps> = ({ children }) => {
+export const ProviderContextProvider: FunctionComponent<ProviderContextProviderProps> = ({
+  children,
+  networks: supportedChainInfoObjects,
+  defaultChainId,
+}) => {
   const [providerType, setProviderType] = useState<SIGNER_TYPE>(SIGNER_TYPE.IDENTITY);
-  const [provider, setProvider] = useState<providers.Provider | Signer>(getProvider);
-  const [account, setAccount] = useState<string>();
+  const [isConnected, setIsConnected] = useState<boolean>();
+  const [currentChainId, setCurrentChainId] = useState<ChainId | undefined>(defaultChainId);
 
-  const initializeMetaMaskSigner = async () => {
-    const { ethereum, web3 } = window;
+  const isSupportedNetwork = (chainId: ChainId) =>
+    supportedChainInfoObjects.findIndex((chainInfoObj) => chainInfoObj.chainId === chainId) > -1;
```

Comment:
```suggestion
  const isSupportedNetwork = (chainId: ChainId) =>
    supportedChainInfoObjects.some((chainInfoObj) => chainInfoObj.chainId === chainId);
```
will this work the same as what is intended?

Reply:
Yup, they should be the same. Better readability with this too. Updated in c4bd2961.